themes/default/content/blog/changes-to-import/index.md

```diff
@@ -0,0 +1,235 @@
+---
+title: "Improving the `pulumi import` experience"
+date: 2022-03-11
+meta_desc: Recent improvements to `pulumi import` make it an even smoother way to import existing cloud infrastructure into Pulumi.
+meta_image: meta.png
+authors:
+    - fraser-waters
+    - david-flanagan
+tags:
+    - features
+    - migration
+    - import
+---
+
+Last year, we [introduced a new feature]({{< relref "blog/pulumi-import-generate-iac-for-existing-cloud-resources" >}}) for Pulumi that allowed you to import existing infrastructure into your Pulumi program. Not only did it bring the resource into the Pulumi state file, but it could generate the source code for your Pulumi program too. Today, we're excited to announce that we've listened to feedback and delivered a plethora of updates an fixies to streamline the import experience; to make it **more useful**, **more convenient**, and **more powerful**.
+
+At Pulumi, we understand that many cloud engineers and platform teams around the world don't have the luxury of greenfield projects, more often than not we're stuck with the impossible task of "refactoring" or "migrating" existing projects to more modern stacks to help increase team productivity, velocity, and stability. These projects aren't trivial and we want to make it easier for teams and organizations to bring their infrastructure into a cloud engineering world. Oh, and worry not you lucky greenfielders ... even if you just wanna ClickOps your way through some resources and import them into your program; that's gonna work just fine too; we won't tell if you don't 😉
+
+To help you understand the changes we've made to help you on your journey, let's take a look at some side by sides examples of how import used to work vs. what we're releasing today.
+
+### Example: AWS S3 Bucket
+
+Previously when importing an S3 bucket, we would set the default values for properties like `acl` and `forceDestroy`. The new behavior will not include properties with default values in the generated code, keeping your code more inline with how you'd write it yourself. Artisanal codegen for the artisinal cloud engineer.
+
+#### Code Comparison
+
+This code would have been generated by the older implementation:
+
+```typescript
+const my_bucket = new aws.s3.Bucket("my-bucket", {
+    acl: "private",
+    bucket: "my-bucket-3f85c54",
+    forceDestroy: false,
+}, {
+    protect: true,
+});
+```
+
+This code will be generated by the newer implementation:
+
+```typescript
+const my_bucket = new aws.s3.Bucket("my-bucket", {
+    arn: "arn:aws:s3:::my-bucket-3f85c54",
+    bucket: "my-bucket-3f85c54",
+    hostedZoneId: "Z3BJ6K6RIION7M",
+    requestPayer: "BucketOwner",
+}, {
+    protect: true,
+});
```

Comment:
Curious why we lead with this example - doesn't it look worse than before? Are there any killer examples when the output actually got shorter?

Reply:
I don't think any examples got shorter. So many resource have these automatic but not default values.

Reply:
How about "Pulumi Import Improvements: Generate IaC for ANY Cloud Resource"?